src/DataStructures/ComplexDataVector.hpp

```diff
@@ -0,0 +1,179 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class Data.
+
+#pragma once
+
+#include <complex>
+#include <cstddef>
+#include <functional>  // for std::reference_wrapper
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/VectorImpl.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+
+// IWYU pragma: no_forward_declare ConstantExpressions_detail::pow
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief Stores a collection of complex function values.
+ *
+ * \details Use ComplexDataVector to represent function values on the
+ * computational domain. Note that interpreting the data also requires knowledge
+ * of the points that these function values correspond to.
+ *
+ * A ComplexDataVector holds an array of contiguous data. The ComplexDataVector
+ * can be owning, meaning the array is deleted when the ComplexDataVector goes
+ * out of scope, or non-owning, meaning it just has a pointer to an array.
+ *
+ * Refer to the \ref DataStructuresGroup documentation for a list of other
+ * available types.
+ *
+ * ComplexDataVectors support a variety of mathematical operations that are
+ * applicable to nodal coefficients. In addition to common arithmetic operations
+ * such as elementwise addition, subtraction, multiplication and division,
+ * elementwise operations on blaze vectors of doubles are supported. See
```

Comment:
I don't understand what `elementwise operations on blaze vectors of doubles` are